### PR TITLE
Split client into client-es5 and client-es6

### PR DESCRIPTION
**Pull request by catrope**
Now that MediaWiki is about to support ES6 in client code, we need a
client ruleset for ES6.

- Move common client rules into client-common
- Remove vue-es5 extends from mediawiki, move it to client-es5
- Keep client as an alias for client-es5

### CODE REVIEW

**Comment by edg2s** [comment]
This is issue #274

File: readme.md [hunk 1]
```diff
@@ -15,55 +15,40 @@ Below are some potential recommended uses:
 
 ### A typical front-end project
 This config allows ES5 code and browser native functions. It will complain about ES6+ code and language features.
+It also automatically applies the Vue plugin and Vue-specific rules to `.vue` files.
```
Comment (edg2s) [inline readme.md:18]: Are they Vue rules a generic Wikimedia style guide, or are they specific to how Vue is used in MediaWiki?

Comment (catrope) [reply readme.md:18]: They are generic. The MW-specific rules were already in mediawiki.json, and I kept them there. A few things were specific to ES5 Vue code, and I kept those in vue-es5

**Comment by catrope** [comment]
> This is issue #274

Thanks for pointing this out. I think the structure you lay out that issue is preferable (tell people to do `browser` + `language/es5` instead of `client-es5`, and `browser` + `language/es6` instead of `client-es6`), but I don't think that I can redo my PR with that structure, because there are rules that are specific to the combination of browser+ES5, and I don't think we could put those in either `browser` or `language/es5`.

**Comment by edg2s** [comment]
Cool. I think `client-common` is an improvement on `browser` which is just a synonym of `client`.

File: readme.md [hunk 2]
```diff
@@ -15,55 +15,40 @@ Below are some potential recommended uses:
...
 	"extends": [
-		"wikimedia/client",
+		"wikimedia/client-es5",
 		"wikimedia/jquery"
 	]
 }
 ```
 
 #### MediaWiki
-Code that runs in MediaWiki can use this config. It enforces rules that are specific to the MediaWiki codebase (core and extensions), such as correct documentation of `mw.message` usage. It also automatically applies the Vue plugin and Vue-specific rules to `.vue` files, including MediaWiki-specific Vue rules such as prohibiting ES6 syntax and prohibiting self-closing tags.
```
Comment (jdforrester) [inline readme.md:39]: We should still mention the self-closing tag prohibition.

**Review by jdforrester — APPROVED** [review]

**Pull request merged**